docs/DeveloperGuide.md

```diff
@@ -1,29 +1,173 @@
 # Developer Guide
 
-## Acknowledgements
+## Design & Implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### Ui
+The `Ui` class handles user input and basic output messages. It can
+* Read user input
+* Output various messages to the user
 
-## Design & implementation
+![img.png](imgs/UiClass.png)
+
+### Storage Class
+![img.png](imgs/StorageClass.png)
+
+### Media Classes
+![img.png](imgs/MovieClass.png)
+
+### Commands
+The command component enables users to make changes to their review list. The command word is taken from the first word
+of the user input, and is processed through the `Parser` class. The class diagram shows how the Commands parent class is
+implemented, as well as its extended classes.
+![img.png](imgs/CommandsClass.png)
+
+#### Add
+The `add` command enables users to create new movie or TV Show reviews and add them to their review list. The following 
+is how the `add` command works:
+1. The `Parser` class takes in the user input, and parses this by using its `processUserInput` function.
+2. If the parsed command word is recognised to be the `add` command, the `executeAdd` function is called.
+3. Using the parsed command, `Parser` then determines whether the review to be added is a movie or Tv show. Otherwise, a 
+DukeException error is thrown.
+4. A new variable of type Media parent class, `toAdd`, is then created. The `AddCommand` function is then called to add
+the Movie or Tv Show `toAdd` into the review list, `reviewList`.
+![img_1.png](imgs/AddCommandSequence.png)
+
```

Comment:
There may be missing/extra lifelines here.

![image](https://user-images.githubusercontent.com/53231968/198250938-bf38aeb5-4681-45a8-a7e5-781258d2bb29.png)

![image](https://user-images.githubusercontent.com/53231968/198250986-893d2313-0df6-4cb7-b187-c0ccef049b82.png)